### PR TITLE
[Internal] Patch: Adds a PatchItemStream API with Stream payload

### PR DESCRIPTION
**Pull request by vivekr20**
* This change adds a new internal patch item stream API to allow for directly passing in a Stream payload for the patch.
    * This helps improve interop API patch performance, which normally go through the Batch API for all scenarios.

- [] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 1]
```diff
@@ -784,6 +784,38 @@ public abstract class Container
...
+        /// </returns>
+        /// <remarks>
+        /// https://aka.ms/cosmosdb-dot-net-exceptions#stream-api
+        /// This is to increase performance and prevent the overhead of throwing exceptions. 
+        /// Check the HTTP status code on the response to check if the operation failed.
+        /// </remarks>
+        /// <example>
+        /// <see cref="Container.PatchItemAsync"/>
+        /// </example>
+        public abstract Task<ResponseMessage> PatchItemStreamAsync(
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:813]: Is there a reason this needs to be exposed out to external users?

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs [hunk 2]
```diff
@@ -1283,6 +1283,49 @@ private string ContcatCachedUriWithId(string resourceId)
                 cancellationToken: cancellationToken);
         }
 
+        public Task<ResponseMessage> PatchItemStreamAsync(
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:1286]: Please add tests

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 1 above)
Comment (vivekr20) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:813]: yup, this API is meant to be purely internal, I'll move it to ContainerInternal.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs (hunk 2 above)
Comment (vivekr20) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:1286]: Ack, will od.

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/CosmosItemUnitTests.cs [hunk 3]
```diff
@@ -454,6 +454,47 @@ public async Task PartitionKeyDeleteUnitTest()
             await this.VerifyPartitionKeyDeleteOperation(new Cosmos.PartitionKey(item.pk), "[\"FF627B77-568E-4541-A47E-041EAC10E46F\"]");
         }
 
+        [TestMethod]
+        public async Task VerifyPatchItemStreamOperation()
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/CosmosItemUnitTests.cs:458]: Is there any reason a e2e was not added?

**Pull request merged**